packages/runtime-core/src/componentOptions.ts

```diff
@@ -431,7 +431,7 @@ interface LegacyOptions<
   computed?: C
   methods?: M
   watch?: ComponentWatchOptions
-  provide?: Data | Function
+  provide?: Data
```

Comment:
Why have you removed `Function`?  functions is a valid overload for `provide`

Reply:
I've changed it.  like this

![C@E~L2VC@FV1YLG40HBS@0B](https://user-images.githubusercontent.com/31752986/162619240-1422eea1-8181-4e8c-be49-89602a3045b7.png)
![5%HI{VFM9Q_0EAO35O` 2DP](https://user-images.githubusercontent.com/31752986/162619243-2a8f0195-0f6d-4026-b6e5-d69e22aa5efc.png)